docs/source/components/rewoo-agent.md

```diff
@@ -0,0 +1,194 @@
+In your YAML file, to use the ReWOO Agent as a workflow:
+```yaml
+workflow:
+  _type: rewoo_agent
+  tool_names: [wikipedia_search, current_datetime, code_generation, math_agent]
+  llm_name: nim_llm
+  verbose: true
+  use_tool_schema: true
+```
+
+In your YAML file, to use the ReWOO Agent as a function:
+```yaml
+functions:
+  calculator_multiply:
+    _type: calculator_multiply
+  calculator_inequality:
+    _type: calculator_inequality
+  calculator_divide:
+    _type: aiq_simple_calculator/calculator_divide
+  math_agent:
+    _type: rewoo_agent
+    tool_names:
+      - calculator_multiply
+      - calculator_inequality
+      - calculator_divide
+    description: 'Useful for performing simple mathematical calculations.'
+```
+
+### Configurable Options:
+<ul><li>
+
+`tool_names`: A list of tools that the agent can call. The tools must be functions configured in the YAML file
+</li><li>
+
+`llm_name`: The LLM the agent should use. The LLM must be configured in the YAML file
+</li><li>
+
+`verbose`: Defaults to False (useful to prevent logging of sensitive data). If set to True, the Agent will log input, output, and intermediate steps.
+</li><li>
+
+`include_tool_input_schema_in_tool_description`: Defaults to True. If set to True, the agent will include tool input schemas in tool descriptions.
+</li><li>
+
+`description`: Defaults to "ReWOO Agent Workflow". When the ReWOO Agent is configured as a function, this config option allows us to control the tool description (for example, when used as a tool within another agent).
+</li><li>
+
+`planner_prompt`: Optional. Allows us to override the planner prompt for the ReWOO Agent. The prompt must have variables for tools and must instruct the LLM to output in the ReWOO planner format.
+</li><li>
+
+`solver_prompt`: Optional. Allows us to override the solver prompt for the ReWOO Agent. The prompt must have variables for plan and task.
+</li><li>
+
+`max_history`:  Defaults to 15. Maximum number of messages to keep in the conversation history.
+</li><li>
+
+`use_openai_api`: Defaults to False.  If set to True, the ReWOO Agent will output in OpenAI API spec. If set to False, strings will be used.
+</li><li>
+
+`additional_instructions`: Optional. Default to None. Additional instructions to provide to the agent in addition to the base prompt.
+</li></ul>
+
+
+## How the ReWOO Agent works
+
+A **ReWOO (Reasoning WithOut Observation) Agent** is an AI system that separates the reasoning process from external observations. Instead of interleaving reasoning and tool calls, it first creates a complete plan and then executes it. This decoupled architecture provides several key advantages:
+
+1. **Token Efficiency**: By planning all steps upfront and using placeholders (e.g., "#E1", "#E2") for intermediate results, ReWOO significantly reduces token consumption. These placeholders are replaced with actual values during execution, eliminating the need to include full tool outputs in each reasoning step.
```

Comment:
Documentation team is recommending to use bullet lists instead of numbered lists in situations where order isn't important.